lib/ash/reactor/builders/transaction.ex

```diff
@@ -0,0 +1,149 @@
+defimpl Reactor.Dsl.Build, for: Ash.Reactor.Dsl.Transaction do
+  @moduledoc false
+
+  alias Reactor.{Builder, Dsl.Build}
+  import Ash.Reactor.BuilderUtils
+  import Reactor.Utils
+
+  @doc false
+  @impl true
+  def build(transaction, reactor) do
+    sub_reactor = Builder.new({Ash.Reactor.TransactionStep, transaction.name})
+    # force the sub-reactor to hot be hooked.
+    sub_reactor = %{sub_reactor | context: Map.put(sub_reactor.context, :__ash_hooked__, true)}
```

Comment:
Why no hooks on the sub_reactor?

Reply:
Because we don't want the subreactor delivering notifications on it's own